invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -343,7 +343,9 @@ def get_search_url():
 def new_record():
     """Create an empty record with default values."""
     record = dump_empty(RDMRecordSchema)
-    record["files"] = {"enabled": True}
+    record["files"] = {
+        "enabled": current_app.config.get("APP_RDM_DEPOSIT_DEFAULT_FILES_ENABLED")
```

Comment:
I think this should be at the record service level, ping @ntarocco and @slint to confirm
potentially it can be improved later

Reply:
Naming was I think is too long, maybe removing the `DEPOSIT`. On the other hand, this would **only** affect UI and people could overpass it via REST since then it would be enabled by default (see https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/components/files.py#L39). Which leads to Karolina's conclusion, you can set it at service-component level (and then the variable naming without `DEPOSIT` makes sense. Full circle hehe

Reply:
Agree, it could be something similar to what we have [for the download-all-files link](https://github.com/slint/invenio-rdm-records/blob/4c9237e90af0dd3c2ddb50a22ac393356f13081a/invenio_rdm_records/resources/config.py#L181), which uses the `FromConfig` descriptor.

I feel though that we will soon run into the classic "config granularity" discussion of single-field config vars vs. overriding the entire config class. I'm not sure how we all arrived at the exact same time to need similar functionalities, but I would say the best place to centralize where record files behaviour config is consumed would be the [`FileOptionsComponent`](https://github.com/inveniosoftware/invenio-records-resources/blob/58503e2a6fdc7bd110f7d3bf3a51e7ca05fd63f0/invenio_records_resources/services/records/components.py#L82). Either that or we merge/inherit this class in [`DraftFilesComponent`](https://github.com/slint/invenio-drafts-resources/blob/3e6d6b5d289ff88e9e094e3ff419cb5fd7b6cdd6/invenio_drafts_resources/services/records/components/files.py#L21) and call it a day.

Maybe we go ahead though with the individual implementation and refactor/merge later on just so that we have something working first.